### PR TITLE
Allow HTML in custom email token sent via API

### PR DESCRIPTION
**Pull request by escopecz**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [n]
| New feature/enhancement? (use the a.x branch)      | [enhancement]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Use case:

A user has an e-shop and wants to send an abandoned cart email. Mautic would need to know about the abandoned cart in order to replace the token correctly. However, the e-shop may send the Mautic email via API and provide the abandoned cart content as a custom token.

Sending custom tokens with email send is already possible:

https://developer.mautic.org/#send-email-to-contact

However, if the token contains some HTML like `custom <b>value</b>` then it will be sent in the email body like `custom &#60;b&#62;value&#60;/b&#62;`. This PR changes the sanitization of the custom token values to allowing HTML characters.

#### Steps to test this PR:

The API call is tested in the modified functional test so I don't think it needs any special user testing, but if anyone wants to go ahead:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Use https://developer.mautic.org/#send-email-to-contact to send an email with custom tokens. Check the PHPUNIT test if you'll need some help setting it up.

### CODE REVIEW

File: app/bundles/EmailBundle/Controller/Api/EmailApiController.php [hunk 1]
```diff
@@ -122,64 +122,65 @@ public function sendAction(Request $request, $id)
...
-                $cleanTokens[$token] = $value;
-            }
+        $cleanTokens = [];
 
-            $leadFields = array_merge(['id' => $leadId], $lead->getProfileFields());
-            // Set owner_id to support the "Owner is mailer" feature
-            if ($lead->getOwner()) {
-                $leadFields['owner_id'] = $lead->getOwner()->getId();
+        foreach ($tokens as $token => $value) {
+            $value = InputHelper::html($value);
```
Comment (escopecz) [inline app/bundles/EmailBundle/Controller/Api/EmailApiController.php:162]: This is the main change in this file. The rest is just change of indentation due to "return early" change.

**Comment by Mancy** [comment]
i would send custom token with value starting by "html://" which will then processed as html inside the email template by changing line in this file EmailApiController.php which is located inside the following path `app/bundles/EmailBundle/Controller/Api/`
you can execute the following from the terminal

`
sed -i "s/\$value = InputHelper::clean(\$value);/if (stripos(\$value, 'html:\/\/') === 0) { \$value = substr(\$value, 7); } else { \$value = InputHelper::clean(\$value); }/" EmailApiController.php
`
or 

you can replace these lines in the sendLeadAction function
```
            foreach ($tokens as $token => $value) {
                $value = InputHelper::clean($value);
                if (!preg_match('/^{.*?}$/', $token)) {
                    $token = '{'.$token.'}';
                }
                
                $cleanTokens[$token] = $value;
            }  
```
to
```
            foreach ($tokens as $token => $value) {
                if (stripos($value, 'html://') === 0) { $value = substr($value, 7); } else { $value = InputHelper::clean($value); }
                if (!preg_match('/^{.*?}$/', $token)) {
                    $token = '{'.$token.'}';
                }

                $cleanTokens[$token] = $value;
            }
```

**Comment by escopecz** [comment]
I like your thinking, but what would it improve? It will be more complicated for users and there is no security benefit as users can use the `html://` prefix for all and any values.

**Review by kuzmany — DISMISSED** [review]

**Review by kuzmany — DISMISSED** [review]

**Comment by RCheesley** [comment]
@Mancy did you have any thoughts on the follow up discussions?

**Comment by RCheesley** [comment]
@escopecz we have a conflict to resolve here, and if it's possible to give a quick example for testers I'm sure it'd get RTC quickly (I'm guessing most wouldn't know where to start with checking unit tests :) ) will also need rebasing to 5.1.

**Review by kuzmany — APPROVED** [review]
Re-approve 👍

**Review by fedys — APPROVED** [review]
Thanks @escopecz!

**Pull request merged**